view/templates/posted_date_widget.tpl

```diff
@@ -30,21 +30,23 @@ function dateSubmit(dateurl) {
 		<li id="posted-date-selector-year-{{$y}}" class="tool">
 			<a class="datebrowse-link" href="#" onclick="openClose('posted-date-selector-{{$y}}'); return false;">{{$y}}</a>
 		</li>
-		<div id="posted-date-selector-{{$y}}" style="display: none;">
-			<ul class="posted-date-selector-months datebrowse-ul">
+		<li id="posted-date-selector-{{$y}}" class="tool posted-date-selector-months" style="display: none;">
+			<ul class="datebrowse-ul">
 				{{foreach $arr as $d}}
 				<li class="tool">
 					<a class="datebrowse-link" href="#" onclick="dateSubmit('{{$url}}/{{$d.1}}/{{$d.2}}'); return false;">{{$d.0}}</a>
 				</li>
 				{{/foreach}}
 			</ul>
-		</div>
+		</li>
 		{{/foreach}}
 		{{if $cutoff}}
+		</ul>
 		</div>
 		<ul class="datebrowse-ul">
 			<li onclick="showHideDates(); return false;" id="posted-date-collapse" class="fakelink tool">{{$showmore}}</li>
 		</ul>
-		{{/if}}
+		{{else}}
```

Comment:
Are you sure with the ```else```?

Reply:
What do you mean with "Are you sure"?

This template (originally taken from hubzilla) is very confusing. If we reach the `$cutoff_year` the original `ul` is closed and a new `div` + a new `ul` is opened. We need to close it at the end. If there is no `$cutoff` we just need to close the original `ul`.

I did several test + html validator tests. And after 2h of testing and try to understand the code I think this PR is the solution.

Reply:
Yeah, the ```<ul>``` does look balanced - but there is another problem with the ```<div>``` that is unbalanced.